src/common/proxy_wasm/ngx_proxy_wasm_maps.c

```diff
@@ -702,6 +705,39 @@ ngx_proxy_wasm_maps_get_response_status(ngx_wavm_instance_t *instance,
 }
 
 
+static ngx_int_t
+ngx_proxy_wasm_maps_set_response_status(ngx_wavm_instance_t *instance,
+    ngx_str_t *value, ngx_proxy_wasm_map_type_e map_type)
+{
+    ngx_uint_t                status;
+    ngx_proxy_wasm_exec_t    *pwexec;
+    ngx_http_wasm_req_ctx_t  *rctx;
+
+    rctx = ngx_http_proxy_wasm_get_rctx(instance);
+
+    status = ngx_atoi(value->data, value->len);
+
+    if (status < 100 || status > 599) {
+        /**
+         * Ideally we would like to return NGX_ERROR here, but the
+         * proxy-wasm-rust-sdk causes a panic when the host returns an error
+         * code. So we ignore invalid values here. This allows the application
+         * code to gracefully check success by setting then getting.
+         */
+        pwexec = ngx_proxy_wasm_instance2pwexec(instance);
+
+        ngx_proxy_wasm_log_error(NGX_LOG_WARN, pwexec->log, 0,
+                                 "ignoring attempt to set invalid response "
+                                 "status: \"%V\"", value);
```

Comment:
@hishamhm Here I narrowed the status range as per Nginx does in other places, and used the same error message for the template. I think a warning here is better than nothing?

Reply:
sounds good to me!